### PR TITLE
Add functions for getting quay repos and robot accounts

### PR DESCRIPTION
**Pull request by tnevrlka**
Add GetAllRepositories and GetAllRobotAccounts functions to quay.go for use in https://github.com/redhat-appstudio/e2e-tests/pull/421.

### CODE REVIEW

**Review by tkdchen — APPROVED** [review]

**Review by mmorhun — CHANGES_REQUESTED** [review]
It will not get all repositories, you need to implement pagination.

File: pkg/quay/quay.go [hunk 1]
```diff
@@ -275,3 +275,79 @@ func (c *QuayClient) AddPermissionsToRobotAccount(organization, imageRepository,
...
+
+	req.Header.Add("Authorization", fmt.Sprintf("%s %s", "Bearer", c.AuthToken))
+	req.Header.Add("Content-Type", "application/json")
+
+	res, err := c.httpClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	if res.StatusCode != 200 {
+		fmt.Printf("error getting repositories, got status code %d", res.StatusCode)
```
Comment (psturc) [inline pkg/quay/quay.go:296]: should this message be added to the error returned below? (i.e. wrapped by `fmt.Errorf()`

and similarly in [the other method](https://github.com/redhat-appstudio/image-controller/pull/19/files#diff-d4a36e5f4e14c5636dfed4031b618cc2a45afc79934eabb0336a1adde773fd1bR334) as well

**Review by mmorhun — APPROVED** [review]
LGTM

**Review by psturc — COMMENTED** [review]
if there's some reasonable way to test the pagination, I'd add such test, otherwise lgtm

**Pull request merged**